src/scheduled_tasks.rs

```diff
@@ -59,24 +70,42 @@ pub fn setup(db_url: String, user_agent: String) -> Result<(), LemmyError> {
   }
 }
 
-/// Reindex the aggregates tables every one hour
-/// This is necessary because hot_rank is actually a mutable function:
-/// https://dba.stackexchange.com/questions/284052/how-to-create-an-index-based-on-a-time-based-function-in-postgres?noredirect=1#comment555727_284052
-fn reindex_aggregates_tables(conn: &mut PgConnection, concurrently: bool) {
-  for table_name in &[
-    "post_aggregates",
-    "comment_aggregates",
-    "community_aggregates",
-  ] {
-    reindex_table(conn, table_name, concurrently);
-  }
-}
-
-fn reindex_table(conn: &mut PgConnection, table_name: &str, concurrently: bool) {
-  let concurrently_str = if concurrently { "concurrently" } else { "" };
-  info!("Reindexing table {} {} ...", concurrently_str, table_name);
-  let query = format!("reindex table {concurrently_str} {table_name}");
-  sql_query(query).execute(conn).expect("reindex table");
+/// Update the hot_rank columns for the aggregates tables
+fn update_hot_ranks(conn: &mut PgConnection) {
+  info!("Updating hot ranks...");
+
+  // Only update for the last week of content
+  let last_week = now - diesel::dsl::IntervalDsl::weeks(1);
```

Comment:
So after one week the hot rank always reaches a fixed value like 0? I think theres an edge case if the instance is down for a while, then items older than a week wont have the hot rank set. Probably also need to update the hot rank of all items older than one week to 0 (or whatever the constant).

Reply:
Yes thats probably fine.

Seems like I deleted the chart by accident, should be added back.